docs/src/main/paradox/google-cloud-pub-sub-grpc.md

```diff
@@ -0,0 +1,107 @@
+
+Scala
+: @@snip (/google-cloud-pub-sub-grpc/src/test/scala/docs/scaladsl/ExampleUsage.scala) { #init-mat }
+
+Java
+: @@snip (/google-cloud-pub-sub-grpc/src/test/java/docs/javadsl/ExampleUsageJava.java) { #init-mat }
+
+## Publishing 
+
+We first construct a message and then a request using Google's builders. We declare a singleton source which will go via our publishing flow. All messages sent to the flow are published to PubSub.
+
+Scala
+: @@snip (/google-cloud-pub-sub-grpc/src/test/scala/docs/scaladsl/ExampleUsage.scala) { #publish-single }
+
+Java
+: @@snip (/google-cloud-pub-sub-grpc/src/test/java/docs/javadsl/ExampleUsageJava.java) { #publish-single }
+
+
+Similarly to before, we can publish a batch of messages for greater efficiency.
+
+Scala
+: @@snip (/google-cloud-pub-sub-grpc/src/test/scala/docs/scaladsl/ExampleUsage.scala) { #publish-fast }
+
+Java
+: @@snip (/google-cloud-pub-sub-grpc/src/test/java/docs/javadsl/ExampleUsageJava.java) { #publish-fast }
+
+Finally, to automatically acknowledge messages and send messages to your own sink, once can do the following:
+
+Scala
+: @@snip (/google-cloud-pub-sub-grpc/src/test/scala/docs/scaladsl/ExampleUsage.scala) { #subscribe-auto-ack }
+
+Java
+: @@snip (/google-cloud-pub-sub-grpc/src/test/java/docs/javadsl/ExampleUsageJava.java) { #subscribe-auto-ack }
+
+## Subscribing
+
+To receive message from a subscription, first we create a `StreamingPullRequest` with a FQRS of a subscription and
+a deadline for acknowledgemts in seconds. Google requires that only the first `StreamingPullRequest` has the subscription
+and the deadline set. This connector takes care of that and clears up the subscription FQRS and the deadline for subsequent
+`StreamingPullRequest` messages.
+
+Scala
+: @@snip (/google-cloud-pub-sub-grpc/src/test/scala/docs/scaladsl/ExampleUsage.scala) { #subscribe }
+
+Java
+: @@snip (/google-cloud-pub-sub-grpc/src/test/java/docs/javadsl/ExampleUsageJava.java) { #subscribe }
+
+Here `pollInterval` is the time between `StreamingPullRequest` are sent when there are no messages in the subscription.
+
+## Running the examples
+
+To run the example code you will need to configure a project and pub/sub in google cloud and provide your own credentials.
+
+There is also an @github[ExampleApp](/google-cloud-pub-sub-grpc/src/test/scala/docs/scaladsl/ExampleApp.scala) that can be used
+to test publishing to topics and receiving messages from subscriptions:
+
+Sbt
+:   &#9;
+
+    ```bash
+    env GOOGLE_APPLICATION_CREDENTIALS=/path/to/application/credentials.json sbt
+
+    // receive messages from a subsciptions
+    > google-cloud-pub-sub-grpc/Test/run subscribe <project-id> <subscription-name>
+
+    // publish a single message to a topic
+    > google-cloud-pub-sub-grpc/Test/run publish-single <project-id> <topic-name>
+
+    // continually publish a message stream to a topic
+    > google-cloud-pub-sub-grpc/Test/run publish-stream <project-id> <topic-name>
+    ```
```

Comment:
Possibly add something about the tools used in the integration tests, should be useful for users of Alpakka, as well.